### ISSUE TITLE
Compilation not working for QNX

### ISSUE THREAD

**Issue by pablorcum** [protocolbuffers/protobuf#8487]
**What version of protobuf and what language are you using?**
Version: master
Language: C++

**What operating system (Linux, Windows, ...) and version?**
Host: Windows 10
Cross-compiling for QNX v7.1

**What runtime / compiler are you using (e.g., python version or gcc version)**
QNX SDP 7.1 - QCC v8.3

**What did you do?**
1. Build CMake project for QNX

**What did you expect to see**
Compilation working. It stopped working after #8106 that removed sys/param.h.

**What did you see instead?**
Source code does not compile due to missing `endian.h`.

**Anything else we should know about your project / environment**
It works by replacing
`#include <endian.h>  // __BYTE_ORDER`
with
`#include <sys/platform.h>  // __BYTE_ORDER`
Actually, sys/param.h is not required and sys/platform.h is enough to enable compilation in files `src\google\protobuf\stubs\port.h` and `src\google\protobuf\io\coded_stream.h`.

**Comment by TeBoring** [comment]
Could you add
```
#elif defined(__QNX__)
#include <sys/platform.h> // __BYTE_ORDER
```
in files changed in https://github.com/protocolbuffers/protobuf/pull/8106/files

**Comment by pablorcum** [comment]
Fixed by #8637
